### PR TITLE
doc: add example for listing ./configure flags

### PR DESCRIPTION
**Pull request by nathaniel-mahieu**
Going through and building bitcoind is a great learning process.  This is the only detail I found missing from the guide.  I think this may be common knowledge but its key information for a new user needing to read the build guide.

### CODE REVIEW

**Comment by laanwj** [comment]
I'd use "configure flags" not "compiler flags", to avoid confusion with CFLAGS/CXXFLAGS

**Comment by jonasschnelli** [comment]
`./configure --help` is somehow unrelated to compiling bitcoin. No strong opinion though.

File: doc/build-unix.md [hunk 1]
```diff
@@ -236,3 +236,9 @@ In this case there is no dependency on Berkeley DB 4.8.
 
 Mining is also possible in disable-wallet mode, but only using the `getblocktemplate` RPC
 call not `getwork`.
+
+Additional Configure Flags
+-------------------------
```
Comment (laanwj) [inline doc/build-unix.md]: Nit: one more `-`

**Comment by laanwj** [comment]
> ./configure --help is somehow unrelated to compiling bitcoin. No strong opinion though.

Some other configure options are explicitly listed in the build instructions. I don't think it hurts to tell that there may be other options available than those.

**Pull request merged**